Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java

```diff
@@ -830,4 +832,34 @@ public void onPlayerPickupItem(final org.bukkit.event.entity.EntityPickupItemEve
             }
         }
     }
+
+    private final class CommandSendListener implements Listener {
+        @EventHandler(priority = EventPriority.NORMAL)
+        public void onCommandSend(final PlayerCommandSendEvent event) {
+            User user = ess.getUser(event.getPlayer());
+
+            ArrayList<String> removedCmds = new ArrayList<>(event.getCommands());
+
+            event.getCommands().removeIf(str -> shouldHideFromUser(str, user));
+
+            removedCmds.removeAll(event.getCommands());
+            ess.getLogger().info("Removed commands: " + removedCmds.toString());
```

Comment:
Does this need to be logged each time or just if debug is enabled?

Reply:
Oops, that should just be if debug is enabled. An earlier version of this wrapped this in a `isDebug` check but I must've removed it by mistake.

Reply:
Ok. Looks good otherwise!